lib/conditional.module.ts

```diff
@@ -0,0 +1,24 @@
+import { ModuleMetadata } from '@nestjs/common';
+import { ConfigModule } from './config.module';
+
+export class ConditionalModule {
+  static async registerWhen(
+    module: Required<ModuleMetadata>['imports'][number],
+    condition: string | ((env: NodeJS.ProcessEnv) => boolean),
+  ) {
+    if (typeof condition === 'string') {
+      const key = condition;
+      condition = env => {
+        return env[key]?.toLowerCase() !== 'false';
+      };
+    }
+    await ConfigModule.envVariablesLoaded;
+    return condition(process.env)
```

Comment:
Wondering if we should call `Logger.debug(...)` when `condition` returns false? Just to improve the overall dev experience

Reply:
Good call! Added that in